hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,302 @@
+      Configuration conf,
+      ClientPool<IMetaStoreClient, TException> metaClients,
+      FileIO fileIO,
+      String catalogName,
+      TableIdentifier viewIdentifier) {
+    String dbName = viewIdentifier.namespace().level(0);
+    this.metaClients = metaClients;
+    this.fileIO = fileIO;
+    this.fullName = BaseMetastoreCatalog.fullTableName(catalogName, viewIdentifier);
+    this.database = dbName;
+    this.viewName = viewIdentifier.name();
+    this.maxHiveTablePropertySize =
+        conf.getLong(HIVE_TABLE_PROPERTY_MAX_SIZE, HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public void doRefresh() {
+    String metadataLocation = null;
+    Table table;
+
+    try {
+      table = metaClients.run(client -> client.getTable(database, viewName));
+      HiveViewOperations.validateTableIsIcebergView(table, fullName);
+
+      metadataLocation =
+          table.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+
+    } catch (NoSuchObjectException | NoSuchIcebergViewException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s.%s", database, viewName);
+      }
+    } catch (TException e) {
+      String errMsg =
+          String.format("Failed to get view info from metastore %s.%s", database, viewName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  public void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newView = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    boolean updateHiveView = false;
+    BaseMetastoreTableOperations.CommitStatus commitStatus =
+        BaseMetastoreTableOperations.CommitStatus.FAILURE;
+
+    try {
+      Table view = null;
+
+      try {
+        view = metaClients.run(client -> client.getTable(database, viewName));
+      } catch (NoSuchObjectException nte) {
+        LOG.trace("View not found {}.{}", database, viewName, nte);
+      }
+
+      if (view != null) {
+        if (!view.getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
+          throw new AlreadyExistsException(
+              "Table with same name already exists: %s.%s", database, viewName);
+        }
+
+        // If we try to create the view but the metadata location is already set, then we had a
+        // concurrent commit
+        if (newView
+            && view.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP)
+                != null) {
+          throw new AlreadyExistsException("View already exists: %s.%s", database, viewName);
+        }
+
+        updateHiveView = true;
+        LOG.debug("Committing existing view: {}", fullName);
+      } else {
+        view =
+            newHmsTable(
```

Comment:
When creating the table/view definition in hms, we should be setting something in the original and expanded text fields to indicate that this is an Iceberg view.  This can help if something non-iceberg attempts to load the view definition.

Reply:
Should we read this from metadata properties or add a constant ?